eggo/director.py

```diff
@@ -0,0 +1,273 @@
+# Licensed to Big Data Genomics (BDG) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The BDG licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import boto
+import boto.cloudformation
+import boto.ec2
+from boto.ec2.networkinterface import NetworkInterfaceCollection, NetworkInterfaceSpecification
+import itertools
+import os
+import sys
+import time
+from datetime import datetime
+from fabric.api import local, env, run, execute, prefix, put, open_shell
+from tempfile import mkdtemp
+
+REGION = 'us-east-1'
+LAUNCHER_AMI = 'ami-a25415cb' # RHEL 6.4 x86
+LAUNCHER_INSTANCE_TYPE = 'm3.large'
+CLUSTER_AMI = LAUNCHER_AMI
+AWS_ACCESS_KEY_ID = os.environ['AWS_ACCESS_KEY_ID']
+AWS_SECRET_ACCESS_KEY = os.environ['AWS_SECRET_ACCESS_KEY']
+EC2_KEY_PAIR = os.environ['EC2_KEY_PAIR']
+EC2_PRIVATE_KEY_FILE = os.environ['EC2_PRIVATE_KEY_FILE']
+STACK_NAME = 'bdg-eggo-{key_pair}'.format(key_pair=EC2_KEY_PAIR)
+CLOUDFORMATION_TEMPLATE='conf/cfn-cloudera-us-east-1-public-subnet.template'
+DIRECTOR_CONF_TEMPLATE='conf/aws.conf'
+
+def provision():
+    # create cloud formation stack (VPC etc)
+    cf_conn = create_cf_connection()
+    create_stack(cf_conn, STACK_NAME)
+
+    # create launcher instance
+    conn = create_ec2_connection()
+    launcher_instance = create_launcher_instance(conn, cf_conn)
+
+    # run bootstrap on launcher
+    execute(run_director_bootstrap, hosts=[launcher_instance.ip_address])
+
+def list():
+    conn = create_ec2_connection()
+    print 'Launcher', get_launcher_instance(conn).ip_address
+    print 'Manager', get_manager_instance(conn).ip_address
+    print 'Gateway', get_gateway_instance(conn).ip_address
+    print 'Master', get_master_instance(conn).ip_address
+
+def login():
+    conn = create_ec2_connection()
+    hosts = get_gateway_instance(conn).ip_address
```

Comment:
Do we generally want to login to the gateway or the master?  I thought the gateway is generally a puny node that is used purely to coordinate the cluster provisioning.